### PR TITLE
[v6r13] Fix security hole

### PR DESCRIPTION
**Pull request by atsareg**
Fixes the security hole due to a possibility of the SQL injection
FIX: NotificationDB - escape fields for sorting in getNotifications()
FIX: MySQL.py - treat in detailed way datetime functions in __escapeString()

### CODE REVIEW

**Comment by phicharp** [comment]
Impressive comprehensive list Andrei! I thought you didn't like them ;-)

**Comment by sposs** [comment]
review ok

**Comment by atsareg** [comment]
Acquiring bad habits. The list comprehension is split

**Comment by phicharp** [comment]
I was kind of joking Andrei, but this one looked really cryptic… On the other hand as long as it works! ;-)
Le 4 mars 2015 à 13:57, Andrei Tsaregorodtsev <notifications@github.com> a écrit :

> Acquiring bad habits. The list comprehension is split
> 
> —
> Reply to this email directly or view it on GitHub.
> 

-- 
Dr Philippe Charpentier
J09210, Physics Department, CERN, CH-1211 Genève 23
LHCb Experiment Distributed Computing Coordinator
Office 2-R-007, Tel : +41 22 767 4244 ,  GSM : +41 76 487 0167
mailto:Philippe.Charpentier@cern.ch  http://cern.ch/Philippe.Charpentier

File: Core/Utilities/MySQL.py [hunk 1]
```diff
@@ -463,17 +478,29 @@ def __escapeString( self, myString ):
...
-      for sV in specialValues:
-        if myString.find( sV ) == 0:
-          return S_OK( myString )
+      # Check datetime functions first
+      if myString.strip() == 'UTC_TIMESTAMP()':
+        return S_OK( myString )
+
+      for func in [ 'TIMESTAMPDIFF', 'TIMESTAMPADD' ]:
+        if myString.strip().startswith( '%s(' % func ) and myString.strip().endswith( ')' ):
+          args = myString.strip()[:-1].replace( '%s(' % func, '' ).strip().split(',')
```
Comment (sposs) [inline Core/Utilities/MySQL.py]: Is it safe to remove the ) by taking all the the last character?

Comment (atsareg) [inline Core/Utilities/MySQL.py]: That's why the string is stripped before, it must end with ')' if it is a legal function call, and this is checked in the previous if statement

Comment (sposs) [inline Core/Utilities/MySQL.py]: Fair enough, I saw the strip after looking for the closing )

**Pull request merged**